### PR TITLE
feat: setup pre-commit hook

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes:

- Setup pre-commit hook using husky

## Context:

- Ref : #129

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,6 +39,7 @@
     "eslint-plugin-import": "2.25.4",
     "eslint-plugin-prettier": "4.0.0",
     "eslint-plugin-promise": "6.0.0",
+    "husky": "^7.0.0",
```
Comment (rarkins) [inline package.json:42]:
```suggestion
    "husky": "7.0.4",
```

File: package.json [hunk 2]
```diff
@@ -59,5 +61,8 @@
     "index.js",
     "lib",
     "index.d.ts"
-  ]
+  ],
+  "dependencies": {
+    "pretty-quick": "3.1.3"
```
Comment (rarkins) [inline package.json:66]: Shouldn't this be in devDependencies?

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -38,9 +39,11 @@
     "eslint-plugin-import": "2.25.4",
     "eslint-plugin-prettier": "4.0.0",
     "eslint-plugin-promise": "6.0.0",
+    "husky": "7.0.4",
     "jest": "27.4.7",
     "npm-run-all": "4.1.5",
     "prettier": "2.5.1",
+    "pretty-quick": "3.1.3"
```
Comment (rarkins) [inline package.json:46]:
```suggestion
    "pretty-quick": "3.1.3",
```

**Comment by rarkins** [comment]
D'oh. This should have been a chore: not feat:

**Comment by RahulGautamSingh** [comment]
> D'oh. This should have been a chore: not feat:

Will be careful next time.

**Comment by viceice** [comment]
@RahulGautamSingh this is not a feature 😉 this change now caused unnecessary release bubbling. So next time only mark as feature if it really changes final package. 🤗

**Pull request merged**